### ISSUE TITLE
Initial groundwork

### ISSUE THREAD

**Issue by jojojames** [jojojames/evil-collection#7]
https://github.com/jojojames/evil-integrations/issues/1#issuecomment-341961849

> (13) Setup: to get started, we need to lay down some ground work on how we setup the bindings.
This is how I did it:
```
;; main file
(defun evil-special-modes-init ()
  (interactive)
  (with-eval-after-load 'calendar   (require 'evil-calendar) (evil-calendar-set-keys))
	;;...
	)

;; mode-specific file
(defun evil-calendar-set-keys ()
  (evil-define-key 'motion calendar-mode-map
    "h" 'calendar-backward-day
		;;...
		))
```

> So it's very similar to your evil-integrations.el except it is wrapped in a function call, as per Emacs packaging standard (only requiring a package should not modify Emacs behaviour).

> What do you think?

Looks good to me.

Lets use this thread for other groundwork topics that may come up.

**Comment by Ambrevar** [comment]
Excellent suggestion from @noctuid (https://github.com/Ambrevar/evil-special-modes/issues/4)

Instead of having a list of `(with-eval-after-load`, let's store the list of mode in a customizable variable so that the user can tweak the variable and then call `evil-collection-init`.  There would be need for one function only.

This has another upside: should some users want to override some of our choices regarding generic bindings (e.g. use `gu` instead of `gx`), they can iterate of the list and apply their own functions.

**Comment by Ambrevar** [comment]
I've modified `evil-collection-init` so that it loads the modes from the `evil-collection-mode-list` variable.

**Comment by jojojames** [comment]
Nice, yeah I checked out the commit. LGTM.

**Comment by Ambrevar** [comment]
Fixed: 13e03e987178d65f34aef928cb98a908fc033021
